src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -223,6 +224,24 @@ protected function getNextAvailableJob($queue)
         return $job ? new DatabaseJobRecord((object) $job) : null;
     }
 
+    /**
+     * Get the lock required for popping the next job.
+     *
+     * @return string
+     */
+    protected function getLockForPopping()
+    {
+        $databaseEngine = $this->database->getPdo()->getAttribute(PDO::ATTR_DRIVER_NAME);
+        $databaseVersion = $this->database->getPdo()->getAttribute(PDO::ATTR_SERVER_VERSION);
+
+        if ($databaseEngine == 'mysql' && version_compare($databaseVersion, '8.0.1', '>=') ||
+            $databaseEngine == 'pgsql' && version_compare($databaseVersion, '9.5', '>=')) {
+            return 'FOR UPDATE SKIP LOCKED';
+        }
+
+        return 'FOR UPDATE';
```

Comment:
'FOR UPDATE' does not seem to work in MSSQL while not in a cursor.

Reply:
FOR UPDATE has no effect if you are not in a database transaction.